content/zh/blog/_posts/2020-12-02-dont-panic-kubernetes-and-docker.md

```diff
@@ -0,0 +1,210 @@
+and CRI-O), but Docker was not designed to be embedded inside Kubernetes, and
+that causes a problem. 
+-->
+我们在这里讨论的是两套不同的环境，这就是造成困惑的根源。
+在你的 Kubernetes 集群中，有一个叫做容器运行时的东西，它负责拉取并运行容器镜像。
+Docker 对于运行时来说是一个流行的选择（其他常见的选择包括 containerd 和 CRI-O），
+但 Docker 并非设计用来嵌入到 Kubernetes，这就是问题所在。
+
+<!-- 
+You see, the thing we call “Docker” isn’t actually one thing&mdash;it’s an entire
+tech stack, and one part of it is a thing called “containerd,” which is a
+high-level container runtime by itself. Docker is cool and useful because it has
+a lot of UX enhancements that make it really easy for humans to interact with
+while we’re doing development work, but those UX enhancements aren’t necessary
+for Kubernetes, because it isn’t a human. 
+-->
+你看，我们称之为 “Docker” 的物件实际上并不是一个物件——它是一个完整的技术堆栈，
+它其中一个叫做 “containerd” 的部件本身，才是一个高级容器运行时。
+Docker 既酷炫又实用，因为它有很多用户体验增强功能，这我们做开发工作时简化了操作，
+Kubernetes 用不着用户增强，毕竟它并非人类。
+
+<!-- 
+As a result of this human-friendly abstraction layer, your Kubernetes cluster
+has to use another tool called Dockershim to get at what it really needs, which
+is containerd. That’s not great, because it gives us another thing that has to
+be maintained and can possibly break. What’s actually happening here is that
+Dockershim is being removed from Kubelet as early as v1.23 release, which
+removes support for Docker as a container runtime as a result. You might be
+thinking to yourself, but if containerd is included in the Docker stack, why
+does Kubernetes need the Dockershim?
+-->
+因为这个用户友好的抽象层，Kubernetes 集群不得不引入一个叫做 Dockershim 的工具来访问它真正需要的 containerd。
+这不是一件好事，因为这引入了额外的运维工作量，而且还可能出错。
+实际上正在发生的事情就是：Dockershim 将在不早于 v1.23 版中从 kubelet 中被移除，也就取消对 Docker 容器运行时的支持。
+你心里可能会想，如果 containerd 已经包含在 Docker 堆栈中，为什么 Kubernetes 需要 Dockershim。
+
+<!-- 
+Docker isn’t compliant with CRI, the [Container Runtime Interface](https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/).
+If it were, we wouldn’t need the shim, and this wouldn’t be a thing. But it’s
+not the end of the world, and you don’t need to panic&mdash;you just need to change
+your container runtime from Docker to another supported container runtime.
+-->
+Docker 不兼容 CRI，
+[容器运行时接口](https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/)。
+如果支持，我们就不需要这个 shim 了，也就没问题了。
+但这也不是世界末日，你也不需要恐慌——你唯一要做的就是把你的容器运行时从 Docker 切换到其他受支持的容器运行时。
+
+<!-- 
+One thing to note: If you are relying on the underlying docker socket
+(`/var/run/docker.sock`) as part of a workflow within your cluster today, moving
+to a different runtime will break your ability to use it. This pattern is often
+called Docker in Docker. There are lots of options out there for this specific
+use case including things like
+[kaniko](https://github.com/GoogleContainerTools/kaniko),
+[img](https://github.com/genuinetools/img), and
+[buildah](https://github.com/containers/buildah). 
+-->
+要注意一点：如果你依赖底层的 docker 套接字(`/var/run/docker.sock`)，作为你集群中工作流的一部分，
```

Comment:
tiny nit:
I would write Docker in title case here
```suggestion
要注意一点：如果你依赖底层的 Docker 套接字(`/var/run/docker.sock`)，作为你集群中工作流的一部分，
```
even though in the original it is lower case.